api/v1/model_types.go

```diff
@@ -46,6 +47,15 @@ type ModelSpec struct {
 	// Will be set from ResourceProfile + Engine if not specified.
 	Image string `json:"image,omitempty"`
 
+	// Service Account to be used by the model pods
```

Comment:
We would like to keep the Model spec as simple as possible. I think it would be better to have these fields defined at the system-level (in the ConfigMap for KubeAI instead of the Model spec). Perhaps these would belong to a field inside of the `internal/config` pkg's `System` struct:

```go
type System struct {
    // ...
    ModelServerPods ModelServerPods `json:"modelServerPods"`
}
```

Let me know if there is any reason why defining the same values across all model Pods would be an issue.

Reply:
What's your thought on this instead?
```golang
type System struct {
         // The service account to use for all model pods
	ModelServiceAccountName string `json:"serviceAccountName,omitempty"`

	// Security Context for the model pods
	ModelPodSecurityContext *corev1.PodSecurityContext `json:"podSecurityContext,omitempty"`

	// Security Context for the model pod containers
	ModelContainerSecurityContext *corev1.SecurityContext `json:"securityContext,omitempty"`
}
```

Reply:
I see and agree. I didn't  get your initial comment.

Reply:
I've made the necessary modifications.